### DISCUSSION TITLE
Dangling xl.meta in data not removed by admin heal --remove

### DISCUSSION THREAD

**Discussion by serrureriehurluberlesque** [minio/minio#20280] [Q&A]
## Expected Behavior
When there is dangling objects, objects that don't have parity on enough disks on the cluster, they should be removed by admin heal --remove

## Current Behavior
In my cluster, I have some objects that I can see only on 1 or 2 disks, I tried to do an admin heal --remove to clean these dangling objects, but without success

## Steps to Reproduce (for bugs)
I don't have steps to reproduce this bug but the setup is like this:
6 servers, one disk for each server

On 2 of ther 6 servers there is in the file system in `[...]/minio/data/somepath/`
`file1/xl.meta`
`file2/xl.meta`
but only on 2 servers of the 6 servers, on the other nothing.

When I do `mc ls somepath/`
Sometimes I see the dangling files, sometimes I don't

And I tried to do:
`mc admin heal -r --remove mycluster/somepath/` but this did nothing

## Context
I think the dangling objects were "created" when one server was stopped during maintenance but there was still some call on the cluster

## Your Environment
minio version RELEASE.2024-06-13T22-53-53Z
Debian GNU/Linux 12 (bookworm)

**Comment by serrureriehurluberlesque** [comment]
I don't understand why [this](https://github.com/minio/minio/issues/20277) has been converted to a discussion, this looks like a bug to me, with minio randomly returning inexistant files and not beeing able to heal them @harshavardhana

**Comment by julienlau** [comment]
are you using tiering ?
